src/evm/Readme.md

```diff
@@ -2,73 +2,71 @@
 
-- `engine` - an instance of `SimulationEngine`
-- `state` - an instance of `SimulationDB` that the `engine` is using
+- `engine` - An instance of `SimulationEngine`
+- `state` - An instance of `EngineDatabaseInterface` that the `engine` is using
 
 ### Default behaviour
 
-In a vanilla engine, if nothing is explicitly done to override a state, simulations happen as if they were executed on
-current block.
-
-Whenever some data needed by the engine is not present in the state, the state queries an Ethereum node for this data.
-This way the engine gets basic information about accounts (such as balance, nonce and code) as well as accounts'
-storage (values stored by smart contracts).
-
-These values are cached until `state.clear_temp_storage()` is called.
+In the default setup, simulations execute as if they were performed on the current block, unless explicitly overridden.
 
 ### Overriding an account
 
-Instead of querying account information or storage from a node, we can set it manually.
-
-This is done with `state.init_account()` method which takes the following arguments:
-
-- account address
-- `AccountInfo` instance
-- permanent storage mapping
-- `mocked` boolean
+Instead of relying on live queries for account data, you can manually configure account information using
+the `state.init_account()` method. This method accepts the following parameters:
 
-`AccountInfo` contains basic account information such as balance, nonce and code (in case of smart contract accounts).
+```
+address: Address,
+account: AccountInfo,
+permanent_storage: Option<HashMap<U256, U256>>,
+mocked: bool,
+```
 
-Permanent storage lets us set some storage slots. If a value of a certain storage slot is needed during simulation, the
-state will first check the permanent storage set here. Only if a slot is not set in the permanent storage, the state
-will query an Ethereum node (but see the next section).
+- `AccountInfo`: Contains core account details like balance, nonce, and code (for smart contract accounts).
+- `permanent_storage`: Allows setting predefined storage slots. When the simulation requires a specific storage slot,
+  the engine first checks this storage.
+- `mocked`: Determines whether the account is mocked (see below).
 
-The next section describes the `mocked` argument.
-
-### Mocked and non-mocked accounts
-
-An overriden account may be set as mocked or non-mocked.
-
-For non-mocked accounts (`mocked=false`), missing storage slots will be queried from a node.
+### Simulation-time overrides
 
-For mocked accounts (`mocked=true`), nothing will be queried from a node. Missing slots have value of 0: if a certain
-storage slot is needed during simulation, but has neither been set in the account's permanent storage (see the previous
-section) nor in the simulation-time overrides (see the next section), then the value of such slot will be 0.
+You can override the chain state only for the duration of a single simulation. This is done by
```

Comment:
The word 'only' there means we don't support any other type of overrides, but we do... 'permanent_storage' above is a permanent override.
```suggestion
You can override the chain state for the duration of a single simulation. This is done by
```